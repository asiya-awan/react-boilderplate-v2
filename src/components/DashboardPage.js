import React from 'react';

const DashboardPage = () => (
  <div>
    Dashobard page content
  </div>
);

export default DashboardPage;
