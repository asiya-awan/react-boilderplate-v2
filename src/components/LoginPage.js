import React from 'react';
import {startLogin} from '../actions/auth';
import { connect } from 'react-redux';

export const LoginPage = ({startLogin}) => (
    <div className="box-layout">  
        <div className="box-layout__box">
            <h1 className=".box-layout__title">Boilerplate App</h1>  
            <p>Tag line for App.</p> 
            <button className="button" value='Login with Google' onClick={startLogin}>Login with Google</button>
        </div>  
    </div>
    
);
const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps) (LoginPage);