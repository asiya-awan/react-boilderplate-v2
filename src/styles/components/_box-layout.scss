.box-layout {
    background: url(/images/bg.jpg) no-repeat center center fixed; 
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
    height: 100vh;
    width: 100vw;
    display: flex;
    justify-content: center;

}

.box-layout__box {
    margin: auto;
    width: 25rem;
    background: fade-out(white, 0.15);
    padding: $l-size $m-size;
    text-align: center;
    border-radius: 3px;
}

.box-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1;

}

.box-layout__button {
    margin: 0 0 $m-size 0;
    line-height: 1;
    
}