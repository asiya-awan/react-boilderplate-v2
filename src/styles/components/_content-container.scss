.content-container {
    max-width: 160rem;
    margin: 0 auto;
    padding:0 $m-size;
}