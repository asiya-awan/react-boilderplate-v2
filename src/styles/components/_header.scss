.header {
    background: $dark-blue;
}

.header__title{
color:  white;
text-decoration: none;
    h1 {
        margin:0;
    }
}

.header__content {
    align-items: center;
    padding: $s-size;
    display:flex;
    justify-content: space-between;
}