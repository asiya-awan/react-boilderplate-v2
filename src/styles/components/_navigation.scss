.nav{
    list-style: none;
    background: rgb(175, 175, 240);
    color: rgb(91, 26, 165);
    padding:1.2rem;
    border-bottom: 2px solid darken(rgb(91, 26, 165), 10%);
}
.nav-item, a{
    display: inline;
    text-decoration: none;

    margin-right:1rem;
}