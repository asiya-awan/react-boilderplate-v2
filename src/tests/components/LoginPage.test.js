import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/LoginPage';

test('should correctly render login page', () => {
  
    const wrapper = shallow(<LoginPage startLogin={() => {}} startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();

})

test('should call startLogin on button click', () => {
    const startLogin = jest.fn();
 
    const wrapper = shallow (<LoginPage startLogin = {startLogin} startLogout={() => {}}/>);
        wrapper.find('button').simulate('click');
        expect(startLogin).toHaveBeenCalled();

});

