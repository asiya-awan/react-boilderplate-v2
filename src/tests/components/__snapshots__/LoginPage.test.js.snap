// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <h1
      className=".box-layout__title"
    >
      Boilerplate App
    </h1>
    <p>
      Tag line for App.
    </p>
    <button
      className="button"
      onClick={[Function]}
      value="Login with Google"
    >
      Login with Google
    </button>
  </div>
</div>
`;
